lib/ex_aws/s3.ex

```diff
@@ -1253,6 +1266,41 @@ defmodule ExAws.S3 do
     end
   end
 
+  @spec presigned_post(
+          config :: map,
+          bucket :: binary,
+          key :: binary | nil,
+          opts :: presigned_post_opts()
+        ) :: presigned_post_result()
+  def presigned_post(config, bucket, key, opts \\ []) do
+    expires_in = Keyword.get(opts, :expires_in, 3600)
+    {:ok, datetime} = DateTime.now("Etc/UTC")
+    expiration_date = DateTime.add(datetime, expires_in, :second)
+
+    datetime = datetime_to_erlang_time(datetime)
+
+    credential = ExAws.Auth.Credentials.generate_credential_v4("s3", config, datetime)
+
+    policy =
+      build_amz_post_policy(datetime, expiration_date, bucket, credential, opts, key)
+      |> config.json_codec.encode!
+      |> Base.encode64()
+
+    signature = ExAws.Auth.Signatures.generate_signature_v4("s3", config, datetime, policy)
+
+    %{
+      url: build_bucket_url(config, bucket),
+      fields: %{
+        "key" => key,
+        "X-Amz-Algorithm" => "AWS4-HMAC-SHA256",
+        "X-Amz-Credential" => credential,
+        "X-Amz-Date" => ExAws.Auth.Utils.amz_date(datetime),
```

Comment:
I suspect the `amz_date` is a holdover from when this library was originally ported from erlcloud - you can get the same effect as this now using:
`datetime |> DateTime.utc_now() |> DateTime.truncate(:second) |> DateTime.to_iso8601(:basic)` without the need for the `datetime_to_erlang_time` faffing around :)